### PR TITLE
[SMALLFIX] Clean up in the underfs module

### PR DESCRIPTION
**Pull request by HorizonNet**
Minor fixes in the **test** module.

*GlusterFSUnderFileSystemFactory*: Simplified return statement.
*HdfsUnderFileSystem*: Converted field into local variable.
*HdfsUnderFileSystemFactory*: Simplified return statement.
*SwiftDirectClient*: Simplified exception handling.

### CODE REVIEW

File: underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java [hunk 1]
```diff
@@ -283,11 +282,11 @@ public long getSpace(String path, SpaceType type) throws IOException {
     if (mFileSystem instanceof DistributedFileSystem) {
       switch (type) {
         case SPACE_TOTAL:
-          return ((DistributedFileSystem) mFileSystem).getDiskStatus().getCapacity();
+          return mFileSystem.getStatus().getCapacity();
```
Comment (aaudiber) [inline underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java]: This needs to be reverted to keep compatibility with hadoop 1 - could you suppress the warning and add an explanation?

**Comment by aaudiber** [comment]
LGTM modulo one issue with hadoop 1 compatibility

**Comment by HorizonNet** [comment]
@aaudiber Thanks for the comment. I changed it back and left some comments.

**Comment by aaudiber** [comment]
LGTM

**Pull request merged**